code/app/src/main/java/com/example/sigma_blue/context/SelectedEntities.java

```diff
@@ -29,7 +29,16 @@ public int size() {
         return selected.size();
     }
     public void updateEntity(T newE, T oldE) {
-        selected.remove(oldE);
+        this.removeEntity(oldE);
         selected.add(newE);
     }
+    private void removeEntity(T e) {
```

Comment:
So we have to generate a new array list on deletion? ArrayList::remove causes an issue?